app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt

```diff
@@ -46,211 +47,205 @@ fun MediaPage(
     val media = viewModel.uiState
```

Comment:
(for the code above)
```
val mediaType = MediaType.safeValueOf(mediaTypeArg)
viewModel.populateMediaPage(id, mediaType)
```
There's a good chance that the MediaType & ID will be in the hiltViewModel's SaveStateHandle, possibly not if it's not a nav-hosted VM..? So you may be able use it in there to get a uiState internally in the viewModel initialisation rather than passing in params to instantiate a UiState

Reply:
Aye, I had this in mind. I was going to address it when implementing shared element transitions.